src/pyk/kcfg/explore.py

```
@@ -0,0 +1,119 @@
+import logging
+from typing import Any, ContextManager, Dict, Final, Iterable, List, Optional, Tuple
+
+from pyk.cli_utils import BugReport
+from pyk.cterm import CTerm
+from pyk.kast.inner import KApply, KInner, KLabel, KVariable, Subst
+from pyk.kast.manip import flatten_label, free_vars
+from pyk.kore.rpc import KoreClient, KoreServer
+from pyk.kore.syntax import Top
+from pyk.ktool import KPrint
+from pyk.prelude.k import GENERATED_TOP_CELL
+from pyk.prelude.ml import is_top, mlEquals, mlTop
+
+_LOGGER: Final = logging.getLogger(__name__)
+
+
+class KCFGExplore(ContextManager['KCFGExplore']):
+    kprint: KPrint
+    _port: int
+    _kore_rpc: Optional[Tuple[KoreServer, KoreClient]]
+    _bug_report: Optional[BugReport]
+
+    def __init__(self, kprint: KPrint, port: int, bug_report: Optional[BugReport] = None):
+        self.kprint = kprint
+        self._port = port
+        self._bug_report = bug_report
+        self._kore_rpc = None
+
+    def __enter__(self) -> 'KCFGExplore':
+        return self
+
+    def __exit__(self, *args: Any) -> None:
+        self.close()
+
+    def kore_rpc(self) -> Tuple[KoreServer, KoreClient]:
```

Ideally this method should not be part of the public interface. Moreover, after `close` calls to `kore_rpc` should raise an exception.

---

Done.